src/Core/Repositories/Queries/IQueryBuilder.cs

```diff
@@ -0,0 +1,7 @@
+﻿using System;
+
+namespace Foundatio.Repositories.Queries {
+    public interface IQueryBuilder<T> where T : class, new() {
+        T Build(bool supportSoftDeletes = false);
+    }
```

Comment:
This could be moved into the elastic repo. I just thought it was pretty generic.